src/cocoa/toga_cocoa/widgets/detailedlist.py

```diff
@@ -57,6 +57,7 @@ def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
             data = value._impl
         except AttributeError:
             data = TogaData.alloc().init()
+            data.retain()
```

Comment:
Why not add `data.autorelease()` here to release once the calling function takes ownership?

Reply:
I tried that... but based on my experimentation, `autorelease` causes an immediate segfault (same for the other use in TogaData).

Reply:
To clarify, I meant a `retain()` call followed by `autorelease()` so that there are no leftover references once the function call has returned. This would be similar to changes in https://github.com/beeware/toga/pull/1275. Or does this also cause a segfault here?